pipeline/ingestor/ingestor_common/src/main/scala/uk/ac/wellcome/platform/ingestor/common/Indexer.scala

```diff
@@ -0,0 +1,81 @@
+package uk.ac.wellcome.platform.ingestor.common
+
+import com.sksamuel.elastic4s.ElasticDsl._
+import com.sksamuel.elastic4s.requests.bulk.{BulkResponse, BulkResponseItem}
+import com.sksamuel.elastic4s.requests.common.VersionType.ExternalGte
+import com.sksamuel.elastic4s.{ElasticClient, Index, Indexable, Response}
+import grizzled.slf4j.Logging
+import uk.ac.wellcome.elasticsearch.model.{CanonicalId, Version}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+trait Indexer[T] extends Logging {
+  val client: ElasticClient
+
+  implicit val ec: ExecutionContext
+  implicit val indexable: Indexable[T]
+  implicit val id: CanonicalId[T]
+  implicit val version: Version[T]
+  val indexName: Index
+
+  final def index(documents: Seq[T]): Future[Either[Seq[T], Seq[T]]] = {
```

Comment:
You could use [`EitherT[Future, Seq[T], Seq[T]]`](https://typelevel.org/cats/datatypes/eithert.html)?

Reply:
NIce thanks! I didn't even know that it existed. I'm going to do it in the next PR though cause I don't want to wait for another round of Travis if there are no objections